### PR TITLE
feat: maven-repair with automatic trigger of NPEfix after detecting failures

### LINKED ISSUE #374: implement repair attempt in travis directly
using `maven npefix`
and
`maven nopol/astor` to be implemented

### PR DESCRIPTION
**Pull request by cokefh**
addresses #374 for NPEFix

### CODE REVIEW

**Comment by monperrus** [comment]
Rebase against master such that the diff only contains the change in the readme and in the maven plugin.

File: src/maven-repair/src/main/java/com/github/tdurieux/repair/maven/plugin/NPEFixMojo.java [hunk 1]
```diff
@@ -192,7 +196,7 @@ public void execute() throws MojoExecutionException {
             jsonObject.write(writer);
             writer.close();
         } catch (Exception e) {
-            throw new RuntimeException(e);
+            Log.info("We don't break the build for errors during the build, continue.");
```
Comment (monperrus) [inline src/maven-repair/src/main/java/com/github/tdurieux/repair/maven/plugin/NPEFixMojo.java:199]: log the exception, be more explicit, elaborate in 2-3 sentences

File: src/maven-repair/src/main/java/com/github/tdurieux/repair/maven/plugin/NPEFixMojo.java [hunk 2]
```diff
@@ -109,6 +110,9 @@ public void execute() throws MojoExecutionException {
         if (npeTests.isEmpty()) {
             throw new RuntimeException("No failing test with NullPointerException or the NPE occurred outside the source.");
         }
+        else if (npeTests.contains(getFailingTests())) {
```
Comment (monperrus) [inline src/maven-repair/src/main/java/com/github/tdurieux/repair/maven/plugin/NPEFixMojo.java:113]: move at the end

File: src/maven-repair/src/main/java/com/github/tdurieux/repair/maven/plugin/NPEFixMojo.java [hunk 3]
```diff
@@ -192,7 +196,7 @@ public void execute() throws MojoExecutionException {
             jsonObject.write(writer);
```
Comment (monperrus) [inline src/maven-repair/src/main/java/com/github/tdurieux/repair/maven/plugin/NPEFixMojo.java:196]: try should start before the actual repair attempt

File: src/maven-repair/src/main/java/com/github/tdurieux/repair/maven/plugin/NPEFixMojo.java (hunk 1 above)
Comment (cokefh) [reply src/maven-repair/src/main/java/com/github/tdurieux/repair/maven/plugin/NPEFixMojo.java]: Tied to explain a bit more, not sure should I print e.printStackTrace();

File: src/maven-repair/src/main/java/com/github/tdurieux/repair/maven/plugin/NPEFixMojo.java (hunk 2 above)
Comment (cokefh) [reply src/maven-repair/src/main/java/com/github/tdurieux/repair/maven/plugin/NPEFixMojo.java]: Done with here

File: src/maven-repair/src/main/java/com/github/tdurieux/repair/maven/plugin/NPEFixMojo.java (hunk 3 above)
Comment (cokefh) [reply src/maven-repair/src/main/java/com/github/tdurieux/repair/maven/plugin/NPEFixMojo.java:194]: Done here.

**Comment by monperrus** [comment]
No, we should never e.printStacktrace()

The good solution is to write e.toString() in the log.

**Comment by cokefh** [comment]
> No, we should never e.printStacktrace() The good solution is to write e.toString() in the log.

Commit a new fix.

**Comment by cokefh** [comment]
Hello @monperrus, it's ready to merge. Please advise. FH

**Comment by monperrus** [comment]
LGTM. Thanks a lot @cokefh and congrats for your first merged contribution to Repairnator!

**Pull request merged**